curriculum/challenges/english/25-front-end-development/lab-product-landing-page/66d9639a8db48f247aa643d2.md

```diff
@@ -0,0 +1,734 @@
+---
+id: 587d78af367417b2b2512b04
+title: Build a Product Landing Page
+challengeType: 14
+dashedName: build-a-product-landing-page
+demoType: onClick
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. Your product landing page should have a `header` element with a corresponding `id="header"`
+2. You should have an image within the `header` element with a corresponding `id="header-img"` (A logo would make a good image here)
+3. Within the `#header` element, you should have a `nav` element with a corresponding `id="nav-bar"`
+4. You should have at least three clickable elements inside the `nav` element, each with the class `nav-link`
+5. When you click a `.nav-link` button in the `nav` element, you should be taken to the corresponding section of the landing page
+6. You should have an embedded product video with `id="video"`
+7. Your landing page should have a `form` element with a corresponding `id="form"`
+8. Within the form, there should be an `input` field with `id="email"` where you can enter an email address.
+9. The `#email` input field should have placeholder text to let users know what the field is for.
+10. The `#email` input field should use HTML5 validation to confirm that the entered text is an email address
+11. Within the form, there should be a submit `input` with a corresponding `id="submit"`.
+12. When you click the `#submit` element, the email should be submitted to a static page (use this mock URL: `https://www.freecodecamp.com/email-submit`).
+13. The navbar should always be at the top of the viewport.
+14. Your product landing page should have at least one media query.
+15. Your product landing page should utilize CSS flexbox at least once.
+
+**Note:** Be sure to link your stylesheet in your HTML and apply your CSS.
+
+# --hints--
+
+You should have a `header` element with an `id` of `header`.
+
+```js
+const el = document.getElementById('header')
+assert(!!el && el.tagName === 'HEADER')
+```
+
+You should have an `img` element with an `id` of `header-img`.
+
+```js
+const el = document.getElementById('header-img')
+assert(!!el && el.tagName === 'IMG')
+```
+
+Your `#header-img` should be a descendant of the `#header`.
+
+```js
+const els = document.querySelectorAll('#header #header-img')
+assert(els.length > 0)
+```
+
+Your `#header-img` should have a `src` attribute.
+
+```js
+const el = document.getElementById('header-img')
+assert(!!el && !!el.src)
+```
+
+Your `#header-img`’s `src` value should be a valid URL (starts with `http`).
+
+```js
+const el = document.getElementById('header-img')
+assert(!!el && /^http/.test(el.src))
+```
```

Comment:
I am able to pass this test without adding something in the src.

Reply:
It's fixed now.